mono/metadata/w32handle.c

```diff
@@ -96,8 +237,10 @@ mono_w32handle_lookup_data (gpointer handle, MonoW32HandleBase **handle_data)
 		return FALSE;
 
 	offset = SLOT_OFFSET ((gsize) handle);
-	if (private_handles [index][offset].type == MONO_W32HANDLE_UNUSED)
+	if (private_handles [index][offset].type == MONO_W32HANDLE_UNUSED) {
+		assert_handle_is_dead (handle);
```

Comment:
Yeah, in theory we could bundle the type in the ref.
Part of this change is to catch issues related to the lifecycle of the 2 variables.